### PR TITLE
Adst/tozpatron 48/adding new pet

### PR DESCRIPTION
**Pull request by Astralius**
- Modified PetsManagement and PetsController to support adding pets (stub).
- Added fake database to demonstrate interactions.
- Changed Services DI setup from Transient to Singleton to prevent auto-resetting of the mockup database.
- Modified Pets/Add.cshtml for model binder,
- Intoduced HttpPost PetsController.Add method to model-bind form data to a new Pet object.

### CODE REVIEW

File: Toz.Dotnet/Toz.Dotnet.Core/Interfaces/IPetsManagementService.cs [hunk 1]
```diff
@@ -5,10 +5,11 @@ namespace Toz.Dotnet.Core.Interfaces
...
-         bool UpdatePet(Pet pet);
-         bool DeletePet(Pet pet);
-         bool AddPet(Pet pet);
-         Pet GetPet(int id);
+		Pet GetPet(int id);
+        List<Pet> GetAllPets();
+        List<Pet> GetSamplePets();
+        bool UpdatePet(Pet pet);
+        bool DeletePet(Pet pet);
+        bool AddPet(Pet pet);
```
Comment (sebastianpec) [inline Toz.Dotnet/Toz.Dotnet.Core/Interfaces/IPetsManagementService.cs]: CRUD C-Create, R-Read, U-Update, D-Delete

Add can happen but also Create is more reliable name for other layers

File: Toz.Dotnet/Toz.Dotnet.Core/Services/PetsManagementService.cs [hunk 2]
```diff
@@ -79,5 +90,12 @@ public Pet GetPet(int id)
             return null;
         }
 
+        private int getFirstAvailableID()
```
Comment (sebastianpec) [inline Toz.Dotnet/Toz.Dotnet.Core/Services/PetsManagementService.cs]: should be 'int?', what if there is not first.
Also "GetFirstAvaibleId", methods, no matter with what accessors -> PascalCase

**Comment by Astralius** [comment]
@sebastianpec I added code modifications as per your suggestions. Thank you.

File: Toz.Dotnet/Toz.Dotnet/Controllers/PetsController.cs [hunk 3]
```diff
@@ -1,33 +1,58 @@
...
+        public IActionResult Index()
         {
-            return View(_petsManagementService.GetPetsList());
+            return View(_petsManagementService.GetAllPets());
         }
 
-        public ActionResult Edit(int id) 
+        [HttpPost]
+        public IActionResult Add(
+            [Bind("Name, Type, sex, Description, Address")]
```
Comment (Evirth) [inline Toz.Dotnet/Toz.Dotnet/Controllers/PetsController.cs]: I think `sex` should be uppercase. Anyway everything looks ok.

Comment (Astralius) [inline Toz.Dotnet/Toz.Dotnet/Controllers/PetsController.cs]: Good eye, done :)

**Comment by Astralius** [comment]
Approve, anyone? :)

**Pull request merged**